### DISCUSSION TITLE
bulk entry of timesheets via API

### DISCUSSION THREAD

**Discussion by orkun-velocity** [kimai/kimai#4664] [General] [answered]
According to the API docs, the only way to create a timesheet is via `POST /api/timesheets` where we can make only one entry at a time. 
Sample curl goes: 

```
curl --location 'http://localhost:8001/api/timesheets?full=' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'X-AUTH-USER: susan_super' \
--header 'Cookie: PHPSESSID=rhrvhjck27v6hfo1se53aajh09' \
--data '{
        "activity": 143,
        "project": 13,
        "user": 8,
        "tags": [],
        "begin": "2024-02-27T15:26:00+0100",
        "end": "2024-02-27T20:05:00+0100",
        "duration": 16740,
        "description": "test local",
        "rate": 353.4,
        "internalRate": 353.4
}'
```
The issue is when we are calling this for an entire week, the response times go over 2.5 seconds, making it harder to use.

Would there be a way to do a bulk insert a la quick_entry using the API?
![image](https://github.com/kimai/kimai/assets/156099079/5040faa1-0cc6-4459-88f3-770038d51b1b)

**Comment by kevinpapst** [comment, marked as answer]
For the performance: you could help testing https://github.com/kimai/kimai/pull/4637 which will reduce the answer time dramatically.

The other request is definitely worth a feature request.
